zapc/src/meta/report.rs

```diff
@@ -151,6 +248,173 @@ impl Report {
+					decl_kind,
+					ticks(decl_name).fg(ERROR)
+				))
+				.with_label(
+					label(span)
+						.with_message(format!("missing {}", ticks(field).fg(ERROR)))
+						.with_color(ERROR),
+				),
+
+			Self::ExpectedValue {
+				value_span,
+				found,
+				expected_values,
+			} => build(kind, value_span).with_message("invalid value").with_label(
+				label(value_span)
+					.with_message(format!("expected {}, found {}", expected_values, found.fg(ERROR)))
+					.with_color(ERROR),
+			),
+
+			Self::ExpectedValueWrongType {
+				value_span,
+				found_type,
+				expected_values,
+				help,
+			} => {
+				let builder = build(kind, value_span).with_message("invalid value").with_label(
+					label(value_span)
+						.with_message(format!(
+							"expected {}, found {}",
+							expected_values,
+							ticks(found_type).fg(ERROR)
+						))
+						.with_color(ERROR),
+				);
+
+				if let Some(help) = help {
+					builder.with_help(help)
+				} else {
+					builder
+				}
+			}
+
+			Self::ExpectedType {
+				value_span,
+				expected_type,
+				found_type,
+			} => build(kind, value_span).with_message("mismatched types").with_label(
+				label(value_span)
+					.with_message(format!(
+						"expected {}, found {}",
+						expected_type,
+						ticks(found_type).fg(ERROR)
+					))
+					.with_color(ERROR),
+			),
+
+			Self::ExpectedTypeFoundScope {
+				scope_span,
+				found_scope,
+			} => build(kind, scope_span)
+				.with_message(format!("expected type, found scope {}", ticks(found_scope).fg(ERROR)))
+				.with_label(label(scope_span).with_color(ERROR)),
+
+			Self::ExpectedRemoteFoundScope {
+				scope_span,
+				found_scope,
+			} => build(kind, scope_span)
+				.with_message(format!("expected remote, found scope {}", ticks(found_scope).fg(ERROR)))
+				.with_label(label(scope_span).with_color(ERROR)),
+
+			Self::ExpectedPositiveNumber { value_span } => build(kind, value_span)
+				.with_message("expected positive number")
+				.with_label(label(value_span).with_color(ERROR)),
+
+			Self::IncorrectGenericCount {
+				type_span,
+				type_name,
+				generic_spans,
+				generics_optional,
+				expected_count,
+				count,
+			} => {
+				let mut labels = vec![label(type_span).with_color(ERROR).with_message(format!(
+					"expected {}{expected_count} generic {}",
+					if generics_optional { "at most " } else { "" },
+					if expected_count == 1 { "argument" } else { "arguments" }
+				))];
+
+				if count > expected_count {
+					let start = generic_spans[expected_count];
+					let end = generic_spans.last().unwrap();
+
+					let span = start.merge(*end);
+
+					labels.push(
+						label(span)
+							.with_color(INFO)
+							.with_message(if count - expected_count == 1 {
+								"remove this generic argument"
+							} else {
+								"remove these generic arguments"
+							}),
+					);
+				}
+
+				build(kind, type_span)
+					.with_message(format!(
+						"type {} takes {}{expected_count} generic {} but {count} generic {} supplied",
```

Comment:
"generic" should be "generics" when `count > 1`

Reply:
This is already accounted for by the following word: `type x takes 5 generic arguments`